packages/pkp-base/src/lib/pkp-base.ts

```diff
@@ -255,6 +254,18 @@ export class PKPBase<T = PKPBaseDefaultParams> {
     }
   }
 
+  public async authenticate(): Promise<void> {
+    this.validateAuthContext();
+
+    const controllerSessionSigs =
+      (await this.authContext?.client?.getSessionSigs(
+        this.authContext.getSessionSigsProps
+      )) || this.controllerSessionSigs;
+
```

Comment:
can we have an ultimate check to see if `controllerSessionSigs` exists?

```ts
if(!controllerSessionSigs){
  throw new Error('bruh.. no session sigs');
}
```

Reply:
Updated the pr with a new validation method `validateAuthenticate() ` which will throw if `controllerSessionSigs` is not defined after `authenticate` is called.